packages/dds/tree/src/test/feature-libraries/modular-schema/comparison.spec.ts

```diff
@@ -200,6 +209,123 @@ describe("Schema Comparison", () => {
+			for (const { name, schema } of a) {
+				updateTreeSchema(repo1, name, schema);
+			}
+			for (const { name, schema } of b) {
+				updateTreeSchema(repo2, name, schema);
+			}
+			return allowsRepoSuperset(defaultSchemaPolicy, repo1, repo2);
+		};
+
+		const emptyTestTree = {
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: new ObjectNodeStoredSchema(new Map()),
+		};
+		const valueTestTree = {
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: valueAnyTree,
+		};
+		const valueEmptyTestTree = {
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: valueEmptyTree,
+		};
+		const optionalTestTree = {
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: optionalTreeWithoutValue,
+		};
+		const optionalEmptyTestTree = {
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: optionalEmptyTree,
+		};
+		const anyTestTree = {
+			name: brand<TreeNodeSchemaIdentifier>("testTree"),
+			schema: anyTreeWithoutValue,
+		};
+
+		it("Incorporating additional fields should result in a superset", () => {
+			// When repo B has more fields than repo A (with the remaining fields the same), regardless
+			// of whether the additional fields are required or optional, repo B should always be considered
+			// the superset of repo A.
+			testOrder(compareTwoRepo, [[emptyTree], [emptyTree, optionalLocalFieldTree]]);
+			testOrder(compareTwoRepo, [
+				[valueLocalFieldTree],
+				[valueLocalFieldTree, optionalLocalFieldTree],
+			]);
+			testOrder(compareTwoRepo, [[emptyTree], [emptyTree, valueLocalFieldTree]]);
+
+			validateOrdering(
+				compareTwoRepo,
+				[[emptyTree, valueLocalFieldTree], [emptyTree]],
+				Ordering.Subset,
+			);
+
+			// If repo B's field is a sequence, potentially accommodating more fields than repo A, it should be
+			// considered a superset
+			testOrder(compareTwoRepo, [[valueTestTree], [anyTestTree]]);
+			testOrder(compareTwoRepo, [[optionalTestTree], [anyTestTree]]);
+			testOrder(compareTwoRepo, [[valueTestTree, emptyTestTree], [anyTestTree]]);
+		});
+
+		it("Repositories with mismatched fields are not incomparable.", () => {
+			validateOrdering(
+				compareTwoRepo,
+				[[valueTestTree, emptyTree], [anyTestTree]],
+				Ordering.Incomparable,
+			);
+		});
+
+		it("Relaxing a field should result in a superset", () => {
+			testOrder(compareTwoRepo, [
+				[valueTestTree],
+				[emptyTestTree],
+				[optionalTestTree],
+				[anyTestTree],
+			]);
+
+			testOrder(compareTwoRepo, [[optionalEmptyTestTree], [optionalTestTree]]);
+			testOrder(compareTwoRepo, [[valueEmptyTestTree], [optionalTestTree]]);
+			testOrder(compareTwoRepo, [[valueTestTree], [optionalEmptyTestTree]]);
+		});
+
+		it("Some scenarios in which two repositories are considered equal", () => {
+			// When the field schema is `required`, no matter with or without child types, they should be Equal
+			validateOrdering(
+				compareTwoRepo,
+				[[valueTestTree], [valueEmptyTestTree]],
```

Comment:
both the things in this test case seem weird to me. Could you look into them more or can we chat about it if you think you understand why they compare equal?